### PR TITLE
Feat(unit test): warp i18n provider

### PR DESCRIPTION
**Pull request by AugustinMauroy**

### CODE REVIEW

File: test/__mocks__/@testing-library/react.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { IntlProvider } from 'react-intl';
+import messages from '../../../i18n/locales/en.json';
+
+const {
+  render: rtlRender,
+  screen,
+  fireEvent,
+  waitFor,
+  // eslint-disable-next-line no-undef
+} = jest.requireActual('@testing-library/react');
```
Comment (ovflowd) [inline test/__mocks__/@testing-library/react.tsx:11]: Why are you doing this instead of just importing testing-library normally?

**Comment by ovflowd** [comment]
Sorry, what does this have to do with i18n provider? I see that you're "creating" a mock for testing-library. You shouldn't be doing that, as the testing library is designed for Unit Tests.

If you want to make a wrapper for i18n, that's not the way. (Also, this is not a mock; this is more or less what we call a "Proxy" (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy).

You should simply add the i18n provider or a mock for the i18n provider itself when you need to use it.

File: test/__mocks__/@testing-library/react.tsx (hunk 1 above)
Comment (AugustinMauroy) [reply test/__mocks__/@testing-library/react.tsx:11]: Be cause it's do like that on nodejs.dev.

> Personaly I never create a internationalized so I learn/discoverd I18n implementation now.

**Comment by AugustinMauroy** [comment]
> You should simply add the i18n provider or a mock for the i18n provider itself when you need to use it.

How to do this.

**Comment by ovflowd** [comment]
> How to do this.

Sorry, @AugustinMauroy, I'm not in the position of a teacher here. If you don't know how to do this, I'd suggest searching on the web or closing this PR. I don't have time to give you constant 1:1's on your PRs.

I wish I had more time, but I don't have :/

**Comment by manishprivet** [comment]
> How to do this.

@AugustinMauroy See this: https://github.com/nodejs/nodejs.org/blob/cab00b47fd4ddc8b1006e3297b620fa4064ee213/components/Common/Banner/__tests__/index.test.tsx#L32-L34

**Comment by AugustinMauroy** [comment]
I will follow the idea of @manishprivet.

> Thanks for your help

**Pull request closed without merging**